src/ystdlib/error_handling/Result.hpp

```diff
@@ -0,0 +1,65 @@
+#ifndef YSTDLIB_ERROR_HANDLING_RESULT_HPP
+#define YSTDLIB_ERROR_HANDLING_RESULT_HPP
+
+#include <system_error>
+
+#include <outcome/config.hpp>
+#include <outcome/std_result.hpp>
+#include <outcome/success_failure.hpp>
+#include <outcome/try.hpp>
+
+namespace ystdlib::error_handling {
+/**
+ * A convenience alias for Outcome's std_result.
+ *
+ * This alias standardizes error handling across the codebase by defaulting the error type to
+ * `std::error_code`, which interoperates with the ystdlib::error_handling::ErrorCode framework,
+ * making it easier to compose errors and propagate them across different modules and libraries.
+ *
+ * @tparam ReturnType The type returned upon success.
+ * @tparam ErrorType The type returned upon failure.
+ */
+template <typename ReturnType, typename ErrorType = std::error_code>
+using Result = OUTCOME_V2_NAMESPACE::std_result<ReturnType, ErrorType>;
+
+/**
+ * Default return value for ystdlib::error_handling::Result<void> when function succeeds.
+ *
+ * Example:
+ *   auto my_func() -> Result<void> {
+ *     // ...
+ *     return success();
+ *   }
+ */
+[[nodiscard]] inline auto success() -> OUTCOME_V2_NAMESPACE::success_type<void> {
+    return OUTCOME_V2_NAMESPACE::success();
+}
```

Comment:
```suggestion
/**
 * @return A value indicating successful completion of a function that returns a void result (i.e.,
 * `Result<void, E>`).
 */
[[nodiscard]] inline auto success() -> OUTCOME_V2_NAMESPACE::success_type<void> {
    return OUTCOME_V2_NAMESPACE::success();
}
```
- In the doc string, we define "void result" to be `Result<void, E>`. I was thinking of maybe "empty result", but using "void" might be more accurate and explicit.
- Then how about renaming to `void_result_success`?
  - The current name `success` can be confusing since it's directly under the `error_handling` namespace.

@kirkrodrigues what do u think?

Reply:
How about `successful_void_result`?

That said, we would need to use `success(<return-val>)` if we wanted to return a successful result when the error type is constructible from the result type or vice-versa, right?

Reply:
- `successful_void_result` lgtm. The one with `get` is too verbose: `return successful_void_result();` looks more concise and still clear.
- Ideally, yes. But I don't think we have such an urgent need if we stick the error type to `std::error_code` since it's not very possible that the successful result can be constructed from an error code. But I'm ok if you want Bill to add it in this PR as well.

Reply:
Thinking about it again, I kinda feel like `error_handling:success()` is not that confusing. @davidlion, what do you think?